tests/next_tests/unit_tests/conftest.py

```diff
@@ -73,19 +89,20 @@ def pretty_format_and_check(root: itir.FencilDefinition, *args, **kwargs) -> str
         (gtfn_cpu.run_gtfn_imperative, True),
         (gtfn_cpu.run_gtfn_with_temporaries, True),
         (gtfn.format_sourcecode, False),
-        (dace_iterator.run_dace_iterator, True),
-    ],
+    ]
+    + OPTIONAL_PROCESSORS,
     ids=lambda p: next_tests.get_processor_id(p[0]),
 )
 def program_processor(request):
-    return request.param
+    backend, _ = request.param
+    backend_id = next_tests.get_processor_id(backend)
 
+    if skip_cases := next_tests.BACKEND_SKIP_TEST_MATRIX.get(backend_id, []):
+        for marker, skip_mark, msg in skip_cases:
```

Comment:
Remove the redundant `if`
```suggestion
    for marker, skip_mark, msg in next_tests.BACKEND_SKIP_TEST_MATRIX.get(backend_id, []):
```